syntax = "proto3";

//import "google/protobuf/empty.proto";
import "google/protobuf/any.proto";
import "google/protobuf/timestamp.proto";

package alise;

//reply HelloReply
message HelloRequest {
  uint32 message = 1;
}

message HelloReply {
  string message = 1;
  string serialnum = 2;
  string hwversion = 3;
  string swversion = 4;
}

//reply Timestamp
message TimeRequest {
}

message NtpStatus {
  bool isNtpEnabled = 1;
}

message HealthQuery {
  bool query = 1;
}

message Health {
// code bits: 0 - there's some errors
// 1 - adminja working normal
// 2 - core working normal
// 3 - ninja working normal
// 4 - vasya working normal
// 5 - petya working normal
// 8-10 - adminja status:
//     000 - not working (none)
//     001 - not installed
//     010 - starting
//     011 - need to merge settings
//     100 - unknown status
//     101 - stopping
//     110 - stopped
//     111 - unknown status
// 11-13 - core status
// 14-16 - ninja status
// 17-19 - vasya status
// 20-22 - petya status
  uint32 code = 1;
  string Desc = 2;
}  

// PWRIN code bits:
// 0 - PWR2
// 1 - PWR1

message PowerStatus {
  uint32 PWRIN = 1;
  uint32 ResetReq = 2;
}

message PackedMessage {
  string message = 1;
  google.protobuf.Any content = 2;
}
